### DISCUSSION TITLE
cpe小知識

### DISCUSSION THREAD

**Discussion by Cerosop** [1101-GS4719-C/Python-Community-GS4719-C#1658] [General]
不知道這到底合不合法但挺方便的

你可以再報到完的練習時間在她發給你的紙上將好用的程式碼寫下來，像是
ios::sync_with_stdio(0);
cin.tie(0)
這兩個可以加速你程式的讀寫速度，有時候超時了加上這兩行就能通過!

還有map或是pq的自定義compare
cmp = [](int &a, int &b){return a > b;);
priority_queue<int, vector<int>, decltype(cmp)> pq(cmp);

這些東西可以讓你在解題時快速去利用比較好用的資料結構，加速你的解題速度並讓你的腦袋不那麼痛，我覺得超方便的!

**Comment by 1chooo** [comment]
太神啦！還真不知道可以先寫下來，之前在寫資結的 Online Judge 的時候的確每題都會加上
```cpp
ios::sync_with_stdio(0);
cin.tie(0)
```

確實可以加速程式運行，不過大多時候還是自己記憶體沒控好，或是沒搞清楚題目的解題關鍵，一直在 WA⋯⋯

**Comment by v0ouo** [comment]
咦我記得不是這樣嗎:
```
ios_base::sync_with_stdio(0);
cin.tie(0)
```
還是兩個都可以

**Comment by Cerosop** [comment]
好像都可以欸，那時候跑沒報錯，超神奇的

**Comment by lazywulf** [comment]
這兩件事做起來其實不大好喔，競程才用吧。.tie() 方法是使輸入流跟輸出流有綁定，會先強制 flush buffer，再進行輸入。所以用 `cin.tie(0)` 可能會有使用者輸入時突然有輸出噴出來。`ios_base::sync_with_stdio(0);` 則是會使 C++ 與 C 的 I/O 不統一。用的話要記得 iostream 跟 stdio 千萬別混用。不然怎死的都不知道。

基本上除非遇到大型 I/O 不然沒必要吧？
